src/App.tsx

```diff
@@ -15,71 +17,95 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string;
+};
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGood: 'Jam',
+  };
+
+  keyCount: number;
+
+  constructor(props: {}) {
+    super(props);
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+    this.keyCount = 0;
+  }
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  // eslint-disable-next-line no-plusplus
+  setKey = () => this.keyCount++;
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  removeGood = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  render() {
+    const { selectedGood } = this.state;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+    return (
+      <main className="section container">
+        <h1 className="title is-flex is-align-items-center">
+          {
+            selectedGood
+              ? `${selectedGood} is selected`
+              : 'No goods selected'
+          }
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={this.removeGood}
+          />
+        </h1>
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+        <table className="table">
+          <tbody>
+            {goods.map(good => (
+              <tr
+                data-cy="Good"
+                className={cn('', {
+                  'has-background-success-light': good === selectedGood,
+                })}
+                key={this.setKey()}
```

Comment:
sorry, didn't get what did you mean here))
```suggestion
                key={good}
```

Reply:
I define a function that will set a unique ID for key prop. It call everytime when component rerender itself.
I did that, because I need to provide an id. And I decide do that. Is it okay ?

Reply:
Actually, you don't pass any value. Your function returns nothing. 
The key shouldn't be just a random value, it should be connected to an element of an array.
Please fix mentor's comment

Reply:
Got it. Thanks a lot :)